File: Python/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        currentSum = 0
        maxSum = 0

        for i in range(k):
            currentSum += nums[i]
        maxSum = currentSum

        for i in range(k, len(nums)):
            currentSum -= nums[i - k]
            currentSum += nums[i]
            maxSum = max(currentSum, maxSum)

        return maxSum / k

File: Swift/maximum-average-subarray-i.swift
class Solution {
    func findMaxAverage(_ nums: [Int], _ k: Int) -> Double {
        var maxSum = 0
        var currentSum = 0

        for i in 0..<k {
            currentSum += nums[i]
        }
        maxSum = currentSum

        for i in k..<nums.count {
            currentSum -= nums[i - k]
            currentSum += nums[i]
            maxSum = max(currentSum, maxSum)
        }

        return Double(maxSum) / Double(k)
    }
}
